packages/core/src/electron-browser/menu/electron-main-menu-factory.ts

```diff
@@ -71,33 +71,49 @@ export class ElectronMainMenuFactory {
         @inject(MenuModelRegistry) protected readonly menuProvider: MenuModelRegistry,
         @inject(KeybindingRegistry) protected readonly keybindingRegistry: KeybindingRegistry
     ) {
-        preferencesService.onPreferenceChanged(debounce(() => {
-            if (this._menu) {
-                for (const item of this._toggledCommands) {
-                    this._menu.getMenuItemById(item).checked = this.commandRegistry.isToggled(item);
+        preferencesService.onPreferenceChanged(
+            debounce(e => {
+                if (e.preferenceName === 'window.menuBarVisibility') {
+                    this.setMenuBar();
                 }
-                electron.remote.getCurrentWindow().setMenu(this._menu);
-            }
-        }, 10));
+                if (this._menu) {
+                    for (const item of this._toggledCommands) {
+                        this._menu.getMenuItemById(item).checked = this.commandRegistry.isToggled(item);
+                    }
+                    electron.remote.getCurrentWindow().setMenu(this._menu);
+                }
+            }, 10)
+        );
         keybindingRegistry.onKeybindingsChanged(() => {
-            const createdMenuBar = this.createMenuBar();
-            if (isOSX) {
-                electron.remote.Menu.setApplicationMenu(createdMenuBar);
-            } else {
-                electron.remote.getCurrentWindow().setMenu(createdMenuBar);
-            }
+            this.setMenuBar();
         });
     }
 
-    createMenuBar(): Electron.Menu {
-        const menuModel = this.menuProvider.getMenu(MAIN_MENU_BAR);
-        const template = this.fillMenuTemplate([], menuModel);
+    setMenuBar(): void {
+        const createdMenuBar = this.createMenuBar();
         if (isOSX) {
-            template.unshift(this.createOSXMenu());
+            electron.remote.Menu.setApplicationMenu(createdMenuBar);
+        } else {
+            electron.remote.getCurrentWindow().setMenu(createdMenuBar);
+        }
+    }
+
+    createMenuBar(): Electron.Menu | null {
+        let preference = this.preferencesService.get<string>('window.menuBarVisibility');
+        preference = preference ? preference : 'classic';
```

Comment:
When would `window.menuBarVisiblity` be `undefined` given it has a default value?

```suggestion
        const preference = this.preferencesService.get<string>('window.menuBarVisibility') ||  'classic';
```

Reply:
I believe this comment is not addressed, do we need to provide the `default` value?
If we need to provide the default, we can:

```ts
const preference = this.preferencesService.get<string>('window.menuBarVisibility', 'classic');
```